common/setups/rasr/nn_system.py

```diff
@@ -0,0 +1,554 @@
+        tf_flow.config[
+            tf_fwd
+        ].input_map.info_0.tensor_name = "extern_data/placeholders/data/data"
+        tf_flow.config[
+            tf_fwd
+        ].input_map.info_0.seq_length_tensor_name = (
+            "extern_data/placeholders/data/data_dim0_size"
+        )
+
+        tf_flow.config[tf_fwd].output_map.info_0.param_name = "log-posteriors"
+        tf_flow.config[
+            tf_fwd
+        ].output_map.info_0.tensor_name = f"{forward_output_layer}/output_batch_major"
+
+        tf_flow.config[tf_fwd].loader.type = "meta"
+        tf_flow.config[tf_fwd].loader.meta_graph_file = tf_graph_path
+        tf_flow.config[tf_fwd].loader.saved_model_file = checkpoint_path
+
+        tf_flow.config[tf_fwd].loader.required_libraries = returnn_op_path
+
+        return tf_flow
+
+    @staticmethod
+    def add_tf_flow_to_base_flow(
+        base_flow: rasr.FlowNetwork,
+        tf_flow: rasr.FlowNetwork,
+        tf_fwd_input_name: str = "tf-fwd-input",
+    ):
+        """
+        Integrate tf-fwd node into the regular flow network
+
+        :param FlowNetwork base_flow:
+        :param FlowNetwork tf_flow:
+        :param str tf_fwd_input_name: see: get_tf_flow()
+        :rtype: FlowNetwork
+        """
+        assert (
+            len(base_flow.outputs) == 1
+        ), "Not implemented otherwise"  # see hard coded tf-fwd input
+        base_output = list(base_flow.outputs)[0]
+
+        input_name = tf_fwd_input_name
+
+        feature_flow = rasr.FlowNetwork()
+        base_mapping = feature_flow.add_net(base_flow)
+        tf_mapping = feature_flow.add_net(tf_flow)
+        feature_flow.interconnect_inputs(base_flow, base_mapping)
+        feature_flow.interconnect(
+            base_flow, base_mapping, tf_flow, tf_mapping, {base_output: input_name}
+        )
+        feature_flow.interconnect_outputs(tf_flow, tf_mapping)
+
+        return feature_flow
+
+    # -------------------- Setup --------------------
+    def init_system(
+        self,
+        hybrid_init_args: RasrInitArgs,
+        train_data: Dict[str, Union[ReturnnRasrDataInput, OggZipHdfDataInput]],
+        cv_data: Dict[str, Union[ReturnnRasrDataInput, OggZipHdfDataInput]],
+        devtrain_data: Optional[
+            Dict[str, Union[ReturnnRasrDataInput, OggZipHdfDataInput]]
+        ] = None,
+        dev_data: Optional[Dict[str, ReturnnRasrDataInput]] = None,
+        test_data: Optional[Dict[str, ReturnnRasrDataInput]] = None,
+        train_cv_pairing: Optional[List[Tuple[str]]] = None,
+    ):
+        self.hybrid_init_args = hybrid_init_args
+
+        self._init_am(**self.hybrid_init_args.am_args)
+
+        devtrain_data = devtrain_data if devtrain_data is not None else {}
+        dev_data = dev_data if dev_data is not None else {}
+        test_data = test_data if test_data is not None else {}
+
+        self._assert_corpus_name_unique(
+            train_data, devtrain_data, cv_data, dev_data, test_data
+        )
```

Comment:
does this mean cv_data == dev_data is not supportet?

Reply:
Also reordering for consistency reasons
```suggestion
        self._assert_corpus_name_unique(
            train_data, cv_data, devtrain_data, dev_data, test_data
        )
```

Reply:
> does this mean cv_data == dev_data is not supportet?

Unfortunately this is still impossible with the current Training jobs :/